tests/xharness/SimpleTcpListener.cs

```diff
@@ -50,6 +66,55 @@ protected override void Start ()
 				}
 			}
 		}
+		
+		void StartTcpTunnel ()
+		{
+			while (!TunnelInitialized) ; // do nothing until the tunnel is ready
```

Comment:
^ that's CPU heavy, sleep a few ms ?

Reply:
Hmm I think I can do better, this was my very first attempt, I can do a asynchronous call, will require some work here and there, but will make everything nicer. Rather than wait seconds or do a thread.sleep, I can do a async completion source that will return the status and then move on.

Reply:
It does not have to be complex - but saving energy is good and green :are